### ISSUE TITLE
Interesse an bugfixes/neuen features?

### ISSUE THREAD

**Issue by nuukee** [tkrebs/ep3-bs#502]
Hallo Tobias&Co,
wir wüden das System gern für unseren Verein einsetzen. Ein paar Sachen bräuchten wir angepasst - das würde ich gern selbst tun. Bevor ich jetzt die Änderungen lokal mache, wäre es doch vielleicht cool, diese ins Repo zu commiten.
Ich sehe aber, daß es noch diverse offene pull requests gibt.
Besteht Interesse an Fixes/Features aus der Community und wenn ja, wie stellt man sowas am geschicktesten an? Gibt es eine Kommunikationsplattform oder so?
Danke & Gruß,
Sven

**Comment by tkrebs** [comment]
Pull-Requests sind immer gerne gesehen. Aufgrund der Vielzahl an verschiedenen Einstellungsmöglichkeiten und Zuständen des Buchungssystems ist es aber nicht immer einfach, Änderungen vorzunehmen, die an anderer Stelle nicht zu unerwünschten Nebeneffekten bis Fehlern führen. Viele Requests ändern (nachvollziehbarerweise) nur Funktionen für die eigenen Bedürfnisse, ohne das Gesamtsystem zu berücksichtigen. Deshalb sind einige Requests derzeit noch offen.

Ich würde aber in jedem Fall, auch für die eigene Arbeit, einen Fork erstellen und die Änderungen dort committen. Anschließend kann man gezielt einzelne Commits per Pull-Request in die Main-Branch übernehmen.

**Comment by nuukee** [comment]
Hallo Tobias,
danke für die Antwort.
Ich war jetzt soweit etwas Code zu schreiben, aber wie immer ist es doch etwas komplizierter als gedacht ;-)

Unser Use-Case:
Wir brauchen auf jeden Fall, daß alle unsere Nutzer subscription bookings vornehmen können. Das geht ja nur als mindestens Mitarbeiter. In dem Fall werden dann aber keinerlei emails generiert.
Das wollte ich eigentlich konfigurierbar machen (als Mitarbeiter/Admin auch confirmation emails generieren).
Aber ich habe dann gemerkt, daß ein normaler Nutzer und Mitarbeiter/Admins komplett andere Code-Pfade durchlaufen und es nicht so leicht ist, mal ebenso diese emails zu generieren. Sehe ich das richtig?
Hast Du eine Idee wie man das lösen kann bevor ich jetzt Code für die Tonne schreibe?
Danke und Gruß,
Sven

**Comment by tkrebs** [comment]
Ja, normale Benutzer und Mitarbeiter unterscheiden sich auch in der Logik recht stark voneinander.
Ich denke, es ist zu aufwändig und entgegen der Programmlogik, den Use-Case über die Mitarbeiter zu lösen.
Ich würde da eher den Buchungsprozess der normalen Benutzer entsprechend anpassen, obwohl auch das einige Arbeit erfordern dürfte.

**Comment by nuukee** [comment]
Hi,
wieder vielen Dank.
Nach einarbeiten in den Code bin ich zu dem Schluss gekommen, dass es deutlich leichter ist, auch admins emails zukommen zu lassen als den Buchungsprozess anzupassen, da hier komplett andere Formulare verwendet werden.
Ich bereite mal einen Patch vor und schicke ihn :-)
